model/registry/instrumentation.yaml

```diff
@@ -0,0 +1,24 @@
+groups:
+- id: registry.instrumentation
+  prefix: instrumentation
+  type: attribute_group
+  brief: >
+    Describes properties about the library that is being instrumented or that
+    is responsible for performing the instrumentation.
+  attributes:
+    - id: source.repo
+      type: string
+      stability: experimental
+      brief: The repository associated with the InstrumentationScope.
+      examples: [github.com/organization/repo, gitlab.com/group/repo]
+    - id: source.target
+      type: string
+      stability: experimental
+      brief: The build target associated with the InstrumentationScope.
+      examples: ['//in/bazel/path/to/my/folder:target', 'inGradleYourLib']
```

Comment:
This information seems to be an internal implementation detail - why is this important on the end-user telemetry? 
I was hoping to have the artifact name.

Reply:
I think there are two separate things that are of interest:

  1. The _source_ / _identity_ / _provenance_ of the code.
  2. The _distribution channel_ through which it was installed/downloaded.

It is possible for the same original source to be bundled, packaged, distributed, etc. through different distribution channels (e.g. installation through `homebrew` vs `fink` or through `pacman` vs `apt`) ...

For the particular use case that I have in mind, I'm interested more in case 1. I feel like `artifact`, `asset`, etc. are more properties associated with case 2. Both can be useful, but for different purposes.

For the telemetry end user, the build target can be relevant if the same repo contains multiple different variants of the same thing. For example, you could imagine that there is  `fast_debug_build`, `asan_enabled_build`, `optimized_deployment_build` of the component. If there are unusual behaviors or problems that are happening, knowing which of these variants was used could provide some insight regarding what is going on. It also provides useful information to include in bug reports raised to the maintainers of the instrumentation library to identify specifically which component is the one that is being complained about.

Reply:
No longer applicable in the scope of this pull request; moved to https://github.com/open-telemetry/semantic-conventions/pull/1094